### ISSUE TITLE
It is not working on low dimensional data

### ISSUE THREAD

**Issue by zaythedatascientist** [ZJULearning/efanna#2]
When I run EFANNA on Yale face B dataset (2174 dimensions), The accuracy was about 90%.
However, if I pick 2 random dimensional from the same dataset, the accuracy reduces to 0.5%.

I also didn't work on some toy datasets that I generated in 2D (e.g. swiss roll data).
Is there any bugs in the algorithm?

**Comment by CongFu92** [comment]
It's probably the data alignment problem. If you wants to use AVX or SSE2 instruction for acceleration, the data point should be align to dimension of  multiple of 8 or 4. You may try the L2Distance without acceleration.

**Comment by grumbelbart** [comment]
@zaythedatascientist Did you make any progress on that front; I'm currently trying the method on 3D data, with a datacloud of ~1M points, and was unable to get any results so far (efanna_index_buildgraph is running for 3 days now to create 1 kd-tree with 1 iteration).

**Comment by CongFu92** [comment]
please show me the parameters you use........I will try to figure it out.

**Comment by grumbelbart** [comment]
The call is as follows:

  ./efanna_index_buildgraph model_efanna.fvecs model_efanna.graph 1 8 1 30 25 10 10

model_efanna.fvecs is a file with 990998 3D points that are sampled from a 2D surface.

Thanks! Any speedup is appreciated!

**Comment by CongFu92** [comment]
@zaythedatascientist I have fixed the bugs on low dimension data. It should work now both for SIMD instructions and normal distance calculation. 
@grumbelbart I cannot reproduce your problem. I generated 990998 3D points randomly, and used your parameters and built a graph of 96% accuracy within 20 seconds. I guess the most possible issue is that your data file format is not right or there are some wrong numbers in your data file model_efanna.fvecs.
